internal/controllers/rest_controller.go

```diff
@@ -148,16 +166,24 @@ func (rc RESTController) OIDCProviderCallback(w http.ResponseWriter, r *http.Req
 // UserInfo shows user info from the OIDC Provider.
 func (rc RESTController) UserInfo(w http.ResponseWriter, r *http.Request) {
 	// workflow to identify if there is a token present.
-	token, err := rc.getCredentials(r)
+	token, authType, err := rc.getCredentials(r)
 	// if a token is not found, return Forbidden.
 	if authErrors.IsTokenNotFound(err) {
-		rc.forbiddenResponse(w, err)
+		rc.forbidden(w, err)
 		return
 	}
 	userInfo, err := rc.oidcService.Decode(token)
+	if authErrors.IsTokenExpired(err) || authErrors.IsTokenSignatureInvalid(err) {
+		if authType == "HeaderCredentialsHandler" {
+			rc.forbidden(w, err)
+			return
+		}
+		rc.logger.Warn(err)
+		rc.RedirectToOIDCProvider(w, r)
+		return
```

Comment:
The same as above, why do we need this authType check?  I think the redirect needs to happen just if the token is expired, an invalid signature just need to return 403.

Reply:
I'm thinking in another use case, for example, machine-to-machine. So , how the redirect will works in that case, when the user is not a web browser.